pkg/cmd/backuprestore/backupserver_test.go

```diff
@@ -0,0 +1,122 @@
+package backuprestore
+
+import (
+	"context"
+	"errors"
+	"github.com/robfig/cron"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+const validSchedule = "* * * * *"
+
+func TestBackupServer_Validate(t *testing.T) {
+	testCases := []struct {
+		name         string
+		backupServer backupServer
+		expErr       error
+	}{
+		{
+			"BackupServer is disabled",
+			backupServer{enabled: false},
+			nil,
+		},
+		{
+			"BackupServer is disabled and invalid schedule",
+			backupServer{
+				enabled:  false,
+				schedule: "invalid schedule",
+			},
+			nil,
+		},
+		{
+			"BackupServer is disabled and invalid schedule and invalid backup directory",
+			backupServer{
+				enabled:  false,
+				schedule: "invalid schedule",
+				backupOptions: backupOptions{
+					backupDir: "",
+				},
+			},
+			nil,
+		},
+		{
+			"BackupServer is enabled",
+			backupServer{
+				enabled: true,
+			},
+			errors.New("error parsing backup schedule : empty spec string"),
+		},
+		{
+			"BackupServer is enabled and invalid schedule",
+			backupServer{
+				enabled:  true,
+				schedule: "invalid schedule",
+			},
+			errors.New("error parsing backup schedule invalid schedule"),
+		},
+		{
+			"BackupServer is enabled and valid schedule",
+			backupServer{
+				enabled:  true,
+				schedule: validSchedule,
+			},
+			nil,
+		},
+		{
+			"BackupServer is enabled and invalid schedule and invalid backup directory",
+			backupServer{
+				enabled:  true,
+				schedule: "invalid schedule",
+				backupOptions: backupOptions{
+					backupDir: "",
+				},
+			},
+			errors.New("error parsing backup schedule invalid schedule"),
+		},
+		{
+			"BackupServer is enabled and valid schedule and invalid backup directory",
+			backupServer{
+				enabled:  true,
+				schedule: validSchedule,
+				backupOptions: backupOptions{
+					backupDir: "",
+				},
+			},
+			nil,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			actErr := tc.backupServer.Validate()
+			if tc.expErr != nil {
+				require.Contains(t, actErr.Error(), tc.expErr.Error())
+			} else {
+				require.Equal(t, tc.expErr, actErr)
+			}
+		})
+	}
+}
+
+func TestNewBackupServer_scheduleBackup(t *testing.T) {
+	cronSchedule, err := cron.ParseStandard(validSchedule)
+	require.NoError(t, err)
+
+	srvr := &backupServer{
+		schedule:      validSchedule,
+		timeZone:      "",
+		enabled:       true,
+		cronSchedule:  cronSchedule,
+		backupOptions: backupOptions{},
+	}
+
+	err = srvr.scheduleBackup(context.Background(), backUpErMock{})
+	require.NoError(t, err)
```

Comment:
that's very happy path, what's with the error cases and the actual scheduling?

Reply:
yes this is all internal to the `runBackup()` .. I can return them for assertion, but its ugly 

any suggestions ?

Reply:
so the errors of the cron are being testing with the validation. `scheduleBackup()` will never be called in case the schedule is invalid https://github.com/openshift/cluster-etcd-operator/blob/483610a715a42e42802be5a61dd0fde9e87130dd/pkg/cmd/backuprestore/backupserver.go#L84-L88 

What i am testing here, is that the scheduling using the ticker is working as expected .. 

I create a context with timeout of `3 minutes` and assert that the `runBackup` has been invoked at least 3 times 

the cron schedule i used is `* * * * *`, the issue is that the test will need at least 3 minutes to run, there is no cron for second iiuc